tools/tck-api/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala

```diff
@@ -129,18 +131,25 @@ object CypherTCK {
           // filters out scenarios with TCKTags.IGNORE
           val included = pickles.filterNot(tagNames(_) contains TCKTags.IGNORE)
 
-          val includedGroupedByName = included.groupBy(_.getName)
-          val includedGroupedAndSorted = includedGroupedByName.mapValues(_.sortBy(_.getLocation.getLine))
+          val includedWithExtractedName = included.map(p => {
+            p.getName match {
+              case pickleNamePattern1(pickleName, _, null) => (p, pickleName, None)
+              case pickleNamePattern1(pickleName, _, exampleName) => (p, pickleName, Some(exampleName))
+              case _ => (p, p.getName, None)
+            }
+          })
+
+          val includedGroupedByKeywordAndName = includedWithExtractedName.groupBy(t => (t._1.getKeyword, t._2))
+          val includedGroupedAndSorted = includedGroupedByKeywordAndName.mapValues(_.sortBy(_._1.getLocation.getLine))
 
           val featureName = feature.getName
           val scenarios = includedGroupedAndSorted.flatMap {
-            case (_, pickles) =>
-              if(pickles.size > 1)
+            case (("Scenario Outline", _), pickles) =>
                 pickles.zipWithIndex.map {
-                  case (pickle, exampleIndex) => toScenario(categories, featureName, pickle, Some(exampleIndex), featureFile)
+                  case (pickle, exampleIndex) => toScenario(categories, featureName, pickle._2, Some(exampleIndex), pickle._3, pickle._1, featureFile)
                 }
-              else
-                pickles.map(pickle => toScenario(categories, featureName, pickle, None, featureFile))
+            case (("Scenario", _), pickles) =>
+                pickles.map(pickle => toScenario(categories, featureName, pickle._2, None, pickle._3, pickle._1, featureFile))
```

Comment:
pickle._3 (the outline example name - I think) is always None here isn't it. Perhaps replace with None.

Reply:
Could. Yes. But I like to pass it through. It does not bother scenario execution at all. However, the ScenarioFormatChecker see it and [can test for it](https://github.com/opencypher/openCypher/pull/481/files#diff-cccf4151c32c8d59e8d1a5742857c1f0d3ee94a11e5d363da51badd5f4bc53cdR55-R62). If we would set the example to `None` here, a non-outline scenario illegally using the ` #Example: ` marker would remain undiscovered. The API would simple gloss over it. The ScenarioFormatChecker would not see. 

Now, one could discuss if the API should not do all the integrity checks in the first place, why have a separate ScenarioFormatChecker anyway? That is decision I was not involved in, but I reckon a good reason to have the separation is to keep the API rather lean. The API is ran much more frequently – when ever scenarios are parsed in to run compatibility tests. The ScenarioFormatChecker only needs to be ran when scenarios have been changed.

Reply:
Sure, sounds reasonable. I hadn't considered that the example name would make it through the parsing to this point if the test is a Scenario instead of a Scenario Outline.